tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/SelectEnvironmentProviderViewModel.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.Mvvm.Messaging;
+using DevHome.Common.Contracts.Services;
+using DevHome.Common.Environments.Models;
+using DevHome.SetupFlow.Models.Environments;
+using DevHome.SetupFlow.Services;
+using Serilog;
+
+namespace DevHome.SetupFlow.ViewModels.Environments;
+
+public partial class SelectEnvironmentProviderViewModel : SetupPageViewModelBase
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(SelectEnvironmentProviderViewModel));
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    private readonly IComputeSystemService _computeSystemService;
+
+    public ComputeSystemProviderDetails SelectedProvider { get; private set; }
+
+    [ObservableProperty]
+    private bool _areProvidersLoaded;
+
+    [ObservableProperty]
+    private int _selectedProviderIndex;
+
+    [ObservableProperty]
+    private ObservableCollection<ComputeSystemProviderViewModel> _providersViewModels;
+
+    public SelectEnvironmentProviderViewModel(
+        ISetupFlowStringResource stringResource,
+        SetupFlowOrchestrator orchestrator,
+        IComputeSystemService computeSystemService)
+           : base(stringResource, orchestrator)
+    {
+        PageTitle = stringResource.GetLocalized(StringResourceKey.SelectEnvironmentPageTitle);
+        _computeSystemService = computeSystemService;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+    }
+
+    public void LoadProviders()
+    {
+        AreProvidersLoaded = false;
+        Orchestrator.NotifyNavigationCanExecuteChanged();
+
+        Task.Run(async () =>
+        {
+            var providerDetails = await _computeSystemService.GetComputeSystemProvidersAsync();
+
+            _dispatcher.TryEnqueue(() =>
+            {
+                ProvidersViewModels = new();
+
+                foreach (var providerDetail in providerDetails)
+                {
+                    ProvidersViewModels.Add(new ComputeSystemProviderViewModel(providerDetail));
+                }
+
+                AreProvidersLoaded = true;
+            });
+        });
+    }
+
+    protected async override Task OnFirstNavigateToAsync()
+    {
+        CanGoToNextPage = false;
+        LoadProviders();
+
+        // Does nothing, but we need to override this as the base expects a task to be returned.
```

Comment:
nit: comment can be omitted

Reply:
If I update the method above in you last comment then LoadProviders would change to LoadProvidersAsync. And this wouldn't be needed anymore

Reply:
updated